src/controllers/StudentController.js

```diff
@@ -0,0 +1,176 @@
+    id: idSchema,
+    email: emailSchema,
+    picUrl: picSchema,
+    name: nameSchema,
+  });
+
+  let data;
+
+  try {
+    data = bodySchema.parse(request.body);
+  } catch (error) {
+    return response.status(400).json(fromZodError(error));
+  }
+
+  try {
+    const student = await StudentService.add(data);
+    return response.status(201).json(student);
+  } catch (error) {
+    if (error instanceof EntryExists) {
+      return response.status(400).json({ error: error.message });
+    } else {
+      return response
+        .status(500)
+        .json({ error: error.message, errorCode: error.code });
+    }
+  }
+}
+
+async function read(request, response) {
+  if (!['ADMIN', 'VERIFIER'].includes(request.role)) {
+    return response.sendStatus(403);
+  }
+
+  const idSchema = z
+    .string()
+    .min(12)
+    .regex(/\d{5}.{2,5}\d{4}/g, 'Invalid ID Format!')
+    .transform((val) => val.toUpperCase());
+
+  let id;
+
+  try {
+    id = idSchema.parse(request.params.id);
+  } catch (error) {
+    return response.status(400).json(fromZodError(error));
+  }
+
+  const student = await StudentService.read(id);
+
+  if (student) {
+    return response.json({ student });
+  } else {
+    return response.sendStatus(404);
+  }
+}
+
+async function update(request, response) {
+  if (request.role !== 'ADMIN') {
+    return response.sendStatus(403);
+  }
+
+  const idSchema = z
+    .string()
+    .min(12)
+    .regex(/\d{5}.{2,5}\d{4}/g, 'Invalid ID Format!')
+    .transform((val) => val.toUpperCase());
+  const emailSchema = z.string().email();
+  const picSchema = z.string().url();
+  const nameSchema = z.string();
+
+  const bodySchema = z.object({
+    email: emailSchema,
+    picUrl: picSchema,
+    name: nameSchema,
+  });
+
+  let id, data;
+
+  try {
+    id = idSchema.parse(request.params.id);
+    data = bodySchema.parse(request.body);
+  } catch (error) {
+    return response.status(400).json(fromZodError(error));
+  }
+
+  try {
+    const student = await StudentService.update(id, data);
+    return response.json(student);
+  } catch (error) {
+    return response.sendStatus(404);
+  }
+}
+
+async function del(request, response) {
+  if (request.role !== 'ADMIN') {
+    return response.sendStatus(403);
+  }
+
+  const idSchema = z
+    .string()
+    .min(12)
+    .regex(/\d{5}.{2,5}\d{4}/g, 'Invalid ID Format!')
+    .transform((val) => val.toUpperCase());
+
+  let id;
+
+  try {
+    id = idSchema.parse(request.params.id);
+  } catch (error) {
+    return response.status(400).json({ error: 'Invalid ID Format!' });
+  }
+
+  try {
+    await StudentService.del(id);
+    return response.status(204).send();
+  } catch (error) {
+    return response.sendStatus(404);
+  }
+}
+
+async function find(request, response) {
+  const emailSchema = z.string().email();
+
+  let email;
+
+  try {
+    email = emailSchema.parse(request.query.email);
+  } catch (error) {
+    response.status(400).json(fromZodError(error));
```

Comment:
Precisa retornar para finalizar a execução.
```suggestion
    return response.status(400).json(fromZodError(error));
```